src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs

```diff
@@ -1706,6 +1706,11 @@ public async Task<NotificationOutcome> SendDirectNotificationAsync(Notification
             requestUri.Path += "messages/$batch";
             AddToQuery(requestUri, "&direct");
 
+            if (notification.GetType().Name == "FcmNotification")
```

Comment:
nit: looks like we have 3 occurrences of the same logic in this class (meaning casting notification type to GCM). Probably we can extract this to separate private utility method and also add specific comment here why do we need to do such things at all.

Reply:
Done, code extracted to FcmToGcmNotificationTypeCast method.